File: Data Structures & Algorithms/sort-array-by-parity/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    sortArrayByParity(nums: number[]): number[] {
        let left = 0;

        for (let right = 0; right < nums.length; right++) {
            if (nums[right] % 2 === 0) {
                [nums[left], nums[right]] = [nums[right], nums[left]];
                left++;
            }
        }

        return nums;
    }
}

File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        counter = 0
        candidate = None

        for n in nums:
            if counter == 0:
                candidate = n
            
            if n == candidate:
                counter += 1
            else:
                counter -= 1
        
        return candidate
            

            

        